src/cluster.c

```diff
@@ -5988,6 +6031,33 @@ NULL
         sds key = c->argv[2]->ptr;
 
         addReplyLongLong(c,keyHashSlot(key,sdslen(key)));
+    } else if (!strcasecmp(c->argv[1]->ptr,"keyshard") && c->argc == 3) {
+        /* CLUSTER KEYSHARD <key> */
+        int slot = keyHashSlot((char*)c->argv[2]->ptr,
+                                       sdslen(c->argv[2]->ptr));
+
+        clusterNode *node = server.cluster->slots[slot];
+        addReplyArrayLen(c, c->argc-2);
+        addReplyMapLen(c, 2);
+
+        addReplyBulkCString(c, "key");
+        addReplyArrayLen(c, 1);
+        addReplyBulkCString(c,(char*)c->argv[2]->ptr);
```

Comment:
Not sure we nee to reprint the key that was passed in.

Reply:
we are thinking to extend this command for multiple key , that is the reason I was printing key in the reply.